pyccl/halos/profiles/einasto.py

```diff
@@ -41,17 +40,19 @@ class HaloProfileEinasto(HaloProfileMatter):
             'cosmo' to calculate the value from cosmology. Default: 'cosmo'
     """
     __repr_attrs__ = __eq_attrs__ = (
-        "truncated", "alpha", "precision_fftlog", "concentration", "normprof",)
+        "truncated", "alpha", "mass_def", "concentration", "precision_fftlog",
+        "normprof",)
 
     @warn_api(pairs=[("c_M_relation", "concentration")])
-    def __init__(self, *, concentration, truncated=True, alpha='cosmo'):
-        if not isinstance(concentration, Concentration):
-            raise TypeError("concentration must be of type `Concentration`")
-
-        self.concentration = concentration
+    def __init__(self, *, concentration, truncated=True, alpha='cosmo',
+                 mass_def):
         self.truncated = truncated
         self.alpha = alpha
-        super().__init__()
+        super().__init__(mass_def=mass_def, concentration=concentration)
+        mvir = MassDef("vir", "matter")
+        self._to_virial_mass = mass_translator(
+            mass_in=self.mass_def, mass_out=mvir,
+            concentration=self.concentration)
         self.update_precision_fftlog(padding_hi_fftlog=1E2,
```

Comment:
You only need to do this if alpha is cosmo

Reply:
That's ok, doesn't hurt to have it anyway, it's a private attribute and takes no time to declare because it is a nested function. The alternative would be a lot more complicated because we'd have to include checks in `update_parameters`.